rclcpp/include/rclcpp/clock.hpp

```diff
@@ -137,6 +137,46 @@ class Clock
     Duration rel_time,
     Context::SharedPtr context = contexts::get_global_default_context());
 
+  /**
+   * Check if the clock is valid.
+   *
+   * An invalid clock is either a clock that is RCL_CLOCK_UNINITIALIZED or a clock with zero-time.
+   * In other words, an uninitialized clock.
+   *
+   * \return true if clock was or became valid
+   */
+  RCLCPP_PUBLIC
+  bool
+  is_valid();
+
+  /**
+   * Wait for clock to become valid.
+   *
+   * \param timeout the maximum time to wait for.
+   * \param wait_tick_ns the time to wait between each iteration of the wait loop (in nanoseconds).
```

Comment:
nit, this param is gone.

Reply:
Whoops

Reply:
Sorry about that :bow: 

https://github.com/ros2/rclcpp/pull/2040/commits/937c2130e827a6216389e68a0ab9c08d3f3b5e62#diff-f8d8723006f022e91ac9a1e42f3622561b8e671d9508c7b17e3b4ea49b8902d6L156

Reply:
Added the context param too! https://github.com/ros2/rclcpp/compare/937c2130e827a6216389e68a0ab9c08d3f3b5e62..23271a6bd124f9345b86a82441810588d35b5de6